src/Datadog.Trace.ClrProfiler.Managed/Extensions.cs

```diff
@@ -69,44 +69,33 @@ public static object Trace(this Span span, Func<object> func, Action<Exception>
             }
         }
 
-        private static Task TraceTask(Span span, Task task, Action<Exception> onComplete)
+        private static async Task TraceTaskAsync(Task task, Action<Exception> onComplete)
         {
-            return task.ContinueWith(t =>
+            try
             {
-                if (t.IsFaulted)
-                {
-                    onComplete(t.Exception);
-                    throw t.Exception;
-                }
-
-                if (t.IsCompleted)
-                {
-                    onComplete(null);
-                }
-            });
+                await task.ConfigureAwait(false);
+                onComplete(null);
+            }
+            catch (Exception ex)
```

Comment:
Nit: would it make sense to use an exception filter as described in this article https://www.thomaslevesque.com/2015/06/21/exception-filters-in-c-6/ to avoid unwinding the stack? In the general case it wouldn't change much, but it would make crash dumps more useful when tracing is enabled.

Reply:
We're not filtering the exception at all (we want to catch any exception), so an exception filter doesn't make sense here.

Also, I think that article is wrong. Using `throw` (instead of `throw ex`) maintains the exception's original stacktrace. See this other blog post: https://scottdorman.github.io/2007/08/20/Difference-between-quotthrowquot-and-quotthrow-exquot-in-.NET/